### PR TITLE
Added input shape parsing

### PR DESCRIPTION
**Pull request by dgearhart**

### CODE REVIEW

**Comment by dnesbitt61** [comment]
:shipit:

File: src/loki/service.cc [hunk 1]
```diff
@@ -148,6 +150,61 @@ namespace valhalla {
...
+      auto gpx = "";
+      auto geojson = "";
+
+      //we require shape or encoded polyline but we dont know which at first
+      try {
+        //uncompressed shape
+        //we dont need to do this unless we want to add some validation
+        if (input_shape) {
+         for (const auto& latlng : *input_shape)
+            shape.push_back(baldr::Location::FromPtree(latlng.second).latlng_);
```
Comment (kevinkreiser) [inline src/loki/service.cc]: no need to store this. i would just validate that each lon,lat made sense in that for loop

File: valhalla/loki/service.h [hunk 2]
```diff
@@ -31,25 +31,27 @@ namespace valhalla {
...
       prime_server::worker_t::result_t isochrones(boost::property_tree::ptree& request, prime_server::http_request_t::info_t& request_info);
-      prime_server::worker_t::result_t attributes(const boost::property_tree::ptree& request, prime_server::http_request_t::info_t& request_info);
+      prime_server::worker_t::result_t trace_route(boost::property_tree::ptree& request, prime_server::http_request_t::info_t& request_info);
 
       boost::property_tree::ptree config;
       boost::optional<std::string> jsonp;
       std::vector<baldr::Location> locations;
       std::vector<baldr::Location> sources;
       std::vector<baldr::Location> targets;
+      std::list<midgard::PointLL> shape;
```
Comment (kevinkreiser) [inline valhalla/loki/service.h]: this is not needed

File: src/loki/service.cc [hunk 3]
```diff
@@ -304,6 +363,7 @@ namespace valhalla {
       locations.clear();
       sources.clear();
       targets.clear();
+      shape.clear();
```
Comment (kevinkreiser) [inline src/loki/service.cc]: not needed

File: src/loki/service.cc [hunk 4]
```diff
@@ -267,10 +325,11 @@ namespace valhalla {
...
             result = isochrones(request_pt, info);
             break;
-          case ATTRIBUTES:
-            result = attributes(request_pt, info);
+          case TRACE_ATTRIBUTES:
+          case TRACE_ROUTE:
+            result = trace_route(request_pt, info);
             break;
-          default:
+           default:
```
Comment (kevinkreiser) [inline src/loki/service.cc]: typo

**Pull request merged**